# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1321714563 28800
#      Sat Nov 19 06:56:03 2011 -0800
# Node ID 3a2f28c08fbd53517266744804d296b3449a58e2
# Parent  aaefd6b2818846e189a1b1bc786386e1325446a0
wilkinson.m: Use Octave spacing conventions in code.  Add more input validation tests.

* wilkinson.m: Use Octave spacing conventions in code.  Add more input validation tests.

diff --git a/scripts/special-matrix/wilkinson.m b/scripts/special-matrix/wilkinson.m
--- a/scripts/special-matrix/wilkinson.m
+++ b/scripts/special-matrix/wilkinson.m
@@ -30,25 +30,32 @@
 ##         (peda)
 
 function retval = wilkinson (n)
 
   if (nargin != 1)
     print_usage ();
   endif
 
-  if (! (isscalar (n) && (n == fix (n)) && n >= 0))
+  if (! (isscalar (n) && n >= 0 && (n == fix (n))))
     error ("wilkinson: N must be a non-negative integer");
   endif
 
   side = ones (n-1, 1);
   center = abs (-(n-1)/2:(n-1)/2);
   retval = diag (side, -1) + diag (center) + diag (side, 1);
 
 endfunction
 
-%!assert (wilkinson(0), [])
-%!assert (wilkinson(1), 0)
-%!assert (wilkinson(2), [0.5,1;1,0.5])
-%!assert (wilkinson(3), [1,1,0;1,0,1;0,1,1])
-%!assert (wilkinson(4), [1.5,1,0,0;1,0.5,1,0;0,1,0.5,1;0,0,1,1.5])
-%!error (wilkinson())
-%!error (wilkinson(1,2))
+
+%!assert (wilkinson (0), [])
+%!assert (wilkinson (1), 0)
+%!assert (wilkinson (2), [0.5,1;1,0.5])
+%!assert (wilkinson (3), [1,1,0;1,0,1;0,1,1])
+%!assert (wilkinson (4), [1.5,1,0,0;1,0.5,1,0;0,1,0.5,1;0,0,1,1.5])
+
+%% Test input validation
+%!error wilkinson ()
+%!error wilkinson (1,2)
+%!error <N must be a non-negative integer> wilkinson (ones (2))
+%!error <N must be a non-negative integer> wilkinson (-1)
+%!error <N must be a non-negative integer> wilkinson (1.5)
+
